src/clib/pioc_support.c

```diff
@@ -93,19 +94,57 @@ int PIOc_strerror(int pioerr, char *errmsg)
 int PIOc_set_log_level(int level)
 {
 #if PIO_ENABLE_LOGGING
-    char log_filename[NC_MAX_NAME];
-
     /* Set the log level. */
     pio_log_level = level;
+#endif /* PIO_ENABLE_LOGGING */
+    return PIO_NOERR;
+}
 
-    /* Create a filename with the rank in it. */
-    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
-    sprintf(log_filename, "pio_log_%d.txt", my_rank);
+/** Initialize logging
+ * Open log file, if not opened yet, or
+ * increment ref count if already open
+ */
+void pio_init_logging(void )
```

Comment:
What is the reason for doing this in Init_Intercomm or PIOc_Init_Async? Why not just leave the initialization for PIOc_set_log_level()?

The problem is that there are functions that the user might call before init_intracomm or Init_Async. What happens to log output there?

What is the reason for moving the logging code? Is this related to a github issue?